### PR TITLE
Beautify hashtags (Camelcase)

### PR DESCRIPTION
**Pull request by BenGWeeks**
This PR resolves [request damus-io/damus#982](https://github.com/damus-io/damus/issues/982).

In the current version you will see:

8212D2A4-5347-4B4C-8917-18C6ADABBFB3_1_201_a

And this formats it nicely for screenreaders to:

CleanShot 2023-04-28 at 16 30 32

### CODE REVIEW

**Comment by joelklabo** [comment]
LGTM 👍

**Comment by jb55** [comment]
not sure we should be changing the formatting of peoples hashtags

**Comment by jb55** [comment]
oh I see this is an accessibility thing...

File: damus/Util/Hashtags.swift [hunk 1]
```diff
@@ -9,60 +9,70 @@ import Foundation
...
-    "onlyzaps": CustomHashtag.zap,
+    "bitcoin": CustomHashtag(imageName: "bitcoin", display: "Bitcoin", color: Color.orange, offset: -3.0),
+    "bitcoinwallet": CustomHashtag(imageName: "bitcoin", display: "Bitcoin", color: Color.orange, offset: -3.0),
+    "bitcoinadoption": CustomHashtag(imageName: "bitcoin", display: "BitcoinAdoption", color: Color.orange, offset: -3.0),
+    "bitcoinjungle": CustomHashtag(imageName: "bitcoin", display: "BitcoinJungle", color: Color.orange, offset: -3.0),
+    "btc": CustomHashtag(imageName: "bitcoin", display: "BTC", color: Color.orange, offset: -3.0),
+    "coffee": CustomHashtag(imageName: "coffee", display: "Coffee", color: DamusColors.brown, offset: -1.0),
+    "coffeechain": CustomHashtag(imageName: "coffee", display: "CoffeeChain", color: DamusColors.brown, offset: -1.0),
+    "nostr": CustomHashtag(imageName: "nostr", display: "Nostr", color: DamusColors.purple, offset: -2.0),
+    "nostrica": CustomHashtag(imageName: "nostr", display: "Nostrica", color: DamusColors.purple, offset: -2.0),
```
Comment (jb55) [inline damus/Util/Hashtags.swift:34]: let's hold off on adding any new hashtags for now

**Comment by BenGWeeks** [comment]
Yeah, I was hesitant too, a la iOS auto-correct (although original message is unchanged). But if it helps screen-readers seems good for me.

File: damus/Util/Hashtags.swift [hunk 2]
```diff
@@ -9,60 +9,70 @@ import Foundation
...
+    var imageOffset: CGFloat?
+    var imageName: String?
+    
     if let custom_hashtag = custom_hashtags[lowertag] {
-        if let col = custom_hashtag.color {
-            attributedString.foregroundColor = col
-        }
-        
-        let name = custom_hashtag.name
+        displayText = custom_hashtag.display
```
Comment (jb55) [inline damus/Util/Hashtags.swift:52]: let's make this an accessibility option, only camelCase it if it's on (default off)

File: damus/Util/Hashtags.swift [hunk 3]
```diff
@@ -9,60 +9,70 @@ import Foundation
...
+        imageName = nil
+    }
+    
+    var attributedString = AttributedString(stringLiteral: "#\(displayText)")
+    attributedString.link = URL(string: "damus:t:\(displayText)")
+    attributedString.foregroundColor = displayColor
+       
+    var text = Text(attributedString)
+    
+    if let imageName = imageName, let img = UIImage(named: "\(imageName)-hashtag") {
```
Comment (jb55) [inline damus/Util/Hashtags.swift:69]:
```suggestion
    if let imageName, let img = UIImage(named: "\(imageName)-hashtag") {
```

**Review by jb55 — CHANGES_REQUESTED** [review]

File: damus/Util/Hashtags.swift (hunk 3 above)
Comment (BenGWeeks) [reply damus/Util/Hashtags.swift]: Applied, actually just wanted to do `if imageName != nil, let ...`

File: damus/Util/Hashtags.swift (hunk 2 above)
Comment (BenGWeeks) [reply damus/Util/Hashtags.swift]: Done. NB Setting applies to only new posts.

Updated default to false.

**Comment by BenGWeeks** [comment]
Updated @jb55.

**Comment by jb55** [comment]
I do not receive ping here btw

File: damus/Util/Hashtags.swift [hunk 4]
```diff
@@ -9,62 +9,74 @@ import Foundation
...
-    "bitcoin": CustomHashtag.bitcoin,
-    "btc": CustomHashtag.bitcoin,
-    "nostr": CustomHashtag.nostr,
-    "coffee": CustomHashtag.coffee,
-    "coffeechain": CustomHashtag.coffee,
-    "plebchain": CustomHashtag.plebchain,
-    "zap": CustomHashtag.zap,
-    "zaps": CustomHashtag.zap,
-    "zapathon": CustomHashtag.zap,
-    "onlyzaps": CustomHashtag.zap,
```
Comment (jb55) [inline damus/Util/Hashtags.swift:40]: can we just keep this style, change `name` to display name and lowercase it when we need to.

Comment (jb55) [reply damus/Util/Hashtags.swift:40]: nm! ignore this comment lol

Comment (BenGWeeks) [reply damus/Util/Hashtags.swift:40]: This doesn't have the CamelCase text in it though.

File: damus/Util/Hashtags.swift (hunk 1 above)
Comment (BenGWeeks) [reply damus/Util/Hashtags.swift]: specifically want removing?

**Comment by jb55** [comment]
I can't merge this due to conflicts. normally I would fix this myself but I don't have the time to do this anymore, sorry.

**Comment by BenGWeeks** [comment]
> I can't merge this due to conflicts. normally I would fix this myself but I don't have the time to do this anymore, sorry.

No worries, happy to help. Conflicts should now be resolved.

**Pull request closed without merging**